src/scripts/main.js

```diff
@@ -1,3 +1,363 @@
+}
+
+function showMessage(TypeOfMessage) {
+  const winMessage = document.querySelector('.message-win');
+  const loseMessage = document.querySelector('.message-lose');
+  const startMessage = document.querySelector('.message-start');
+
+  switch (TypeOfMessage) {
+    case 'win':
+      winMessage.classList.remove('hidden');
+      loseMessage.classList.add('hidden');
+      startMessage.classList.add('hidden');
+      break;
+    case 'lose':
+      winMessage.classList.add('hidden');
+      loseMessage.classList.remove('hidden');
+      startMessage.classList.add('hidden');
+      break;
+    case 'start':
+      winMessage.classList.add('hidden');
+      loseMessage.classList.add('hidden');
+      startMessage.classList.remove('hidden');
+      break;
+    case 'nothing':
+      winMessage.classList.add('hidden');
+      loseMessage.classList.add('hidden');
+      startMessage.classList.add('hidden');
+      break;
+    default:
+      return 0;
+  }
+}
+
+function notEmptyTile() {
+  for (const row of table) {
+    for (const tile of row) {
+      if (tile === 0) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function notMergeTile() {
+  for (const row of table) {
+    for (let i = 0; i < row.length; i++) {
+      const tile = row[i];
+      const nextTile = row[i + 1];
+
+      if (tile === nextTile) {
+        return false;
+      }
+    }
+  }
+
+  for (let j = 0; j < 4; j++) {
+    for (let k = 0; k < 3; k++) {
+      const tile = table[k][j];
+      const nextRowTile = table[k + 1][j];
+
+      if (tile === nextRowTile) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function moveHorizontally(direction) {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (const row of table) {
+    direction(row);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      generateRandomTile();
+      updateGameFields();
+      break;
+    }
+  }
+}
+
+function moveVertically(direction) {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (let j = 0; j < 4; j++) {
+    direction(
+      [table[0][j], table[1][j], table[2][j], table[3][j]], j
+    );
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      generateRandomTile();
+      updateGameFields();
+      break;
+    }
+  }
+}
+
+document.addEventListener('keydown', e => {
+  switch (e.key) {
+    case 'a': {
+      moveHorizontally(left);
+      break;
+    }
+
+    case 'ArrowLeft': {
+      moveHorizontally(left);
+      break;
+    }
```

Comment:
we can combine these cases
```suggestion
    case 'a': 
    case 'ArrowLeft': {
      moveHorizontally(left);
      break;
    }
```

Reply:
Unfortunately, the linter requires a 'break' after each 'case'. I asked this question in the chat, In this case it is impossible